beacon-chain/p2p/discovery.go

```diff
@@ -70,21 +92,18 @@ func convertToMultiAddr(nodes []*discv5.Node) []ma.Multiaddr {
 	return multiAddrs
 }
 
-func convertToSingleMultiAddr(node *discv5.Node) (ma.Multiaddr, error) {
-	ip4 := node.IP.To4()
+func convertToSingleMultiAddr(node *enode.Node) (ma.Multiaddr, error) {
+	ip4 := node.IP().To4()
 	if ip4 == nil {
 		return nil, errors.New("node doesn't have an ip4 address")
 	}
-	pubkey, err := node.ID.Pubkey()
-	if err != nil {
-		return nil, errors.Wrap(err, "could not get pubkey from node ID")
-	}
+	pubkey := node.Pubkey()
 	assertedKey := convertToInterfacePubkey(pubkey)
 	id, err := peer.IDFromPublicKey(assertedKey)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get peer id")
 	}
-	multiAddrString := fmt.Sprintf("/ip4/%s/tcp/%d/p2p/%s", ip4.String(), node.TCP, id)
+	multiAddrString := fmt.Sprintf("/ip4/%s/tcp/%d/p2p/%s", ip4.String(), node.UDP(), id)
```

Comment:
UDP? This multiaddr is a TCP string

Reply:
currently both udp and tcp connections are under the same port. For discv5 only their UDP port is specified not TCP. So if I use TCP I would get zero. I plan to address this in another PR, where we specify our separate UDP and TCP ports and then appropriately can add them to our multiaddr

Reply:
OK please add a note / comment because this is a bit odd and another engineer may not have that context